cupy/cuda/compiler.py

```diff
@@ -263,6 +273,8 @@ def _compile(
             raise
         return ptx, mapping
 
+    options += (_get_arch_for_options(arch=arch, jitify=jitify),)
```

Comment:
I don't think this should be here? (See my main comment.)

Reply:
the cache key is obtained in preprocess :), if we don't do it here, some compiler calls may fail. (There were some tests failing because of this in `cuda_tests/test_compiler`

Reply:
Is it possible to simplify it as follows?
```python
if backend == 'nvrtc':
    options += (_get_arch_for_options(arch=arch, jitify=jitify),)
```
Perhaps it'd allow you to use it as part of the cache key *and* fix the compiler error without calling `_get_arch_for_options()` twice?

Reply:
sorry I don't follow :),
the tests in `cuda_tests/test_compiler` call `compile_using_nvrtc`, so the cache checking and all that will be bypassed. and we need to check it here. I don't currently see a way to avoid the two calls :(

Reply:
Ah got it, sorry I misread. I understand it now. `_compile_with_cache_cuda()` calls `compile_using_nvrtc()`. I was hoping we just need to fix the options once in the former, and I missed that the latter is used directly in the tests.